From ab5ddf261e2a8c840c0224b5c8ef5932465e4eb8 Mon Sep 17 00:00:00 2001
From: Frantisek Sumsal <frantisek@sumsal.cz>
Date: Thu, 7 Jul 2022 14:12:38 +0200
Subject: [PATCH] test: create an ASan wrapper for `getent` and `su`

since they "suffer" from the same issue as `login` and other binaries
that load PAM stuff

(cherry picked from commit fdb70dd9222219307ca53662e789fc9304ca3616)

Related: #2087652
---
 test/test-functions | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/test-functions b/test/test-functions
index 42ad16050c..8523cf2d21 100644
--- a/test/test-functions
+++ b/test/test-functions
@@ -2410,9 +2410,9 @@ inst_binary() {
 
     # Same as above, but we need to wrap certain libraries unconditionally
     #
-    # login, useradd, userdel - dlopen()s (not only) systemd's PAM modules
+    # getent, login, su, useradd, userdel - dlopen()s (not only) systemd's PAM modules
     # tar - called by machinectl in TEST-25
-    if get_bool "$IS_BUILT_WITH_ASAN" && [[ "$bin" =~ /(login|tar|useradd|userdel)$ ]]; then
+    if get_bool "$IS_BUILT_WITH_ASAN" && [[ "$bin" =~ /(getent|login|su|tar|useradd|userdel)$ ]]; then
         wrap_binary=1
     fi
 
